Modules/Relationships/test/Relationships.Domain.Tests/Tests/Aggregates/Relationships/RelationshipReactivationRequestTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Backbone.BuildingBlocks.Domain;
+using Backbone.Modules.Relationships.Domain.Aggregates.Relationships;
+using Backbone.Modules.Relationships.Domain.Tests.Extensions;
+using Backbone.Tooling;
+using FluentAssertions;
+using Xunit;
+using static Backbone.Modules.Relationships.Domain.Tests.TestHelpers.TestData;
+
+namespace Backbone.Modules.Relationships.Domain.Tests.Tests.Aggregates.Relationships;
+public class RelationshipReactivationRequestTests
+{
+    [Fact]
+    public void Relationship_reactivation_request_creates_an_audit_log_entry()
+    {
+        // Arrange
+        SystemTime.Set("2000-01-01");
+        var relationship = CreateTerminatedRelationship();
+
+        // Act
+        relationship.Reactivate(IDENTITY_2, DEVICE_2);
+
+        // Assert
+        relationship.AuditLog.Should().HaveCount(4);
+
+        var auditLogEntry = relationship.AuditLog.Last();
+
+        auditLogEntry.Id.Should().NotBeNull();
+        auditLogEntry.Reason.Should().Be(RelationshipAuditLogEntryReason.ReactivationRequested);
+        auditLogEntry.OldStatus.Should().Be(RelationshipStatus.Terminated);
+        auditLogEntry.NewStatus.Should().Be(RelationshipStatus.Terminated);
+        auditLogEntry.CreatedBy.Should().Be(IDENTITY_2);
+        auditLogEntry.CreatedByDevice.Should().Be(DEVICE_2);
+        auditLogEntry.CreatedAt.Should().Be(DateTime.Parse("2000-01-01"));
+    }
+
+    [Fact]
+    public void Can_only_request_relationship_reactivation_when_relationship_is_in_status_terminated()
+    {
+        // Arrange
+        var relationship = CreatePendingRelationship();
+
+        // Act
+        var acting = () => relationship.Reactivate(IDENTITY_2, DEVICE_2);
+
+        // Assert
+        acting.Should().Throw<DomainException>().WithError(
+            "error.platform.validation.relationshipRequest.relationshipIsNotInCorrectStatus",
+            nameof(RelationshipStatus.Terminated)
+        );
+    }
+
+    [Fact]
+    public void Cannot_Reactivate_An_Already_Requested_Reactivation()
```

Comment:
Here is this "reactivate a reactivation" again

Reply:
Changes made.